pipeline/backend/local/clone.go

```diff
@@ -0,0 +1,185 @@
+	if e.pluginGitBinary != "" {
+		state.pluginGitBinary = e.pluginGitBinary
+		return nil
+	}
+
+	log.Info().Msg("no global 'plugin-git' installed, try to download for current workflow")
+	state.pluginGitBinary = filepath.Join(state.homeDir, "plugin-git")
+	if runtime.GOOS == "windows" {
+		state.pluginGitBinary += ".exe"
+	}
+	return downloadLatestGitPluginBinary(state.pluginGitBinary)
+}
+
+// execClone executes a clone-step locally
+func (e *local) execClone(ctx context.Context, step *types.Step, state *workflowState, env []string) error {
+	if err := e.setupClone(state); err != nil {
+		return fmt.Errorf("setup clone step failed: %w", err)
+	}
+
+	if err := checkGitCloneCap(); err != nil {
+		return fmt.Errorf("check for git clone capabilities failed: %w", err)
+	}
+
+	if step.Image != constant.DefaultCloneImage {
+		// TODO: write message into log
+		log.Warn().Msgf("clone step image '%s' does not match default git clone image. We ignore it assume git.", step.Image)
+	}
+
+	rmCmd, err := writeNetRC(step, state)
+	if err != nil {
+		return err
+	}
+
+	env = append(env, "CI_WORKSPACE="+state.workspaceDir)
+
+	// Prepare command
+	var cmd *exec.Cmd
+	if runtime.GOOS == "windows" {
+		pwsh, err := exec.LookPath("powershell.exe")
+		if err != nil {
+			return err
+		}
+		cmd = exec.CommandContext(ctx, pwsh, "-Command", fmt.Sprintf("%s ; $code=$? ; %s ; if (!$code) {[Environment]::Exit(1)}", state.pluginGitBinary, rmCmd))
+	} else {
+		cmd = exec.CommandContext(ctx, "/bin/sh", "-c", fmt.Sprintf("%s ; $code=$? ; %s ; exit $code", state.pluginGitBinary, rmCmd))
+	}
+	cmd.Env = env
+	cmd.Dir = state.workspaceDir
+
+	// Get output and redirect Stderr to Stdout
+	e.output, _ = cmd.StdoutPipe()
+	cmd.Stderr = cmd.Stdout
+
+	state.stepCMDs[step.Name] = cmd
+
+	return cmd.Start()
+}
+
+// writeNetRC write a netrc file into the home dir of a given workflow state
+func writeNetRC(step *types.Step, state *workflowState) (string, error) {
+	if step.Environment["CI_NETRC_MACHINE"] == "" {
+		return "", nil
+	}
+
+	file := filepath.Join(state.homeDir, ".netrc")
+	rmCmd := fmt.Sprintf("rm \"%s\"", file)
+	if runtime.GOOS == "windows" {
+		file = filepath.Join(state.homeDir, "_netrc")
+		rmCmd = fmt.Sprintf("del \"%s\"", file)
+	}
+
+	return rmCmd, os.WriteFile(file, []byte(fmt.Sprintf(
+		netrcFile,
+		step.Environment["CI_NETRC_MACHINE"],
+		step.Environment["CI_NETRC_USERNAME"],
+		step.Environment["CI_NETRC_PASSWORD"],
+	)), 0o600)
+}
+
+// downloadLatestGitPluginBinary download the latest plugin-git binary based on runtime OS and Arch
+// and saves it to dest
+func downloadLatestGitPluginBinary(dest string) error {
+	type asset struct {
+		Name               string
+		BrowserDownloadURL string `json:"browser_download_url"`
+	}
+
+	type release struct {
+		Assets []asset
+	}
+
+	// get latest release
+	req, _ := http.NewRequest(http.MethodGet, "https://api.github.com/repos/woodpecker-ci/plugin-git/releases/latest", nil)
```

Comment:
Well yes but it would add that code dep to the local backend ... witch increase bin size

This looked easy to implement ... but I'll checkou the other way now ...